intranet/apps/signage/views.py

```diff
@@ -26,3 +31,95 @@ def signage_display(request, display_id):
     context["sign"] = sign
     context["page_args"] = (sign, request)
     return render(request, "signage/base.html", context)
+
+
+def eighth(request):
+    block_id = request.GET.get("block_id")
+    block_increment = 0
+    internal_ip = check_internal_ip(request)
+    if internal_ip:
+        return internal_ip
+
+    if block_id is None:
+        block = EighthBlock.objects.get_first_upcoming_block()
+        if block is None:
+            last_block = EighthBlock.objects.order_by("date").last()
+            if last_block is not None:
+                block = last_block
+    else:
+        try:
+            block = EighthBlock.objects.prefetch_related("eighthscheduledactivity_set").get(id=block_id)
+        except EighthBlock.DoesNotExist:
+            if EighthBlock.objects.count() == 0:
+                # No blocks have been added yet
+                return render(request, "eighth/display.html", {"no_blocks": True})
+            else:
+                # The provided block_id is invalid
+                raise http.Http404
+
+    if "block_increment" in request.GET:
+        block_increment = request.GET.get("block_increment", 0)
+
+    try:
+        block_increment = int(block_increment)
+    except ValueError:
+        block_increment = 0
+
+    if block_increment > 0:
+        next_blocks = block.next_blocks()
+        if next_blocks.count() >= block_increment:
+            block = next_blocks[block_increment - 1]
+    elif block_increment < 0:
+        index = -block_increment - 1
+        prev_blocks = block.previous_blocks()
+        if prev_blocks.count() > index:
+            block = prev_blocks[index]
+
+    user = User.get_signage_user()
+
+    serializer_context = {"request": request, "user": user}
+    block_info = EighthBlockDetailSerializer(block, context=serializer_context).data
+    try:
+        reload_mins = float(request.GET.get("reload_mins") or 5)
+    except Exception:
+        reload_mins = 5
+
+    touch_signage = not request.GET.get("no_touch")
+
+    next_block = list(block.next_blocks(1))
```

Comment:
Why convert this sliced queryset to a list? You can just take the first element of it on this line + 2.